chat-widget/src/common/utils.ts

```diff
@@ -421,4 +421,20 @@ export const createFileAndDownload = (fileName: string, blobData: string, mimeTy
     document.body.appendChild(aElement);
     aElement.click();
     document.body.removeChild(aElement);
+};
+
+/**
+ * 
+ * Replace placeholders with format {0}..{n} , in a string with values
+ * 
+ * @param template String with placeholders to be replaced
+ * @param values array of values to replace the placeholders
+ * @returns formatted string with replaced values
+ */
+// use of any for values as array of any type is passed
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const  formatTemplateString = (templateMessage : string, values : any) => {
+    return templateMessage.replace(/{(\d+)}/g, (match, index) => {
+        return typeof values[index] !== "undefined" ? values[index] : match;
+    });
 };
```

Comment:
Can be more generic and cleaner using below. Please check

function replacePlaceholders(inputString: string, ...values: any[]): string {
  return inputString.replace(/{(\d+)}/g, (match, index) => {
    const valueIndex = parseInt(index, 10);
    if (valueIndex < values.length) {
      return values[valueIndex];
    } else {
      return match; // If the index is out of range, keep the placeholder as is.
    }
  });
}

Reply:
Ill stick with my version, from my PoV is cleaner and shorter, all scenarios are included in unit tests

if there is a particular scenario Im not testng please let me know